tests/dbt/test_adapter.py

```diff
@@ -58,6 +63,76 @@ def test_adapter_relation(sushi_test_project: Project, runtime_renderer: t.Calla
         == "[]"
     )
 
+    adapter_mock = mocker.MagicMock()
+    adapter_mock.dialect = "snowflake"
+
+    context.target = SnowflakeConfig(
+        account="test",
+        user="test",
+        authenticator="test",
+        name="test",
+        database="test",
+        schema="test",
+    )
+
+    renderer = runtime_renderer(context, engine_adapter=adapter_mock)
+
+    # bla and bob will be normalized to uppercase since we're dealing with Snowflake
+    bla_id = exp.to_identifier("BLA", quoted=False)
+    bob_id = exp.to_identifier("BOB", quoted=False)
+
+    renderer("{{ adapter.get_relation(database=None, schema='bla', identifier='bob') }}")
+    adapter_mock._get_data_objects.assert_has_calls([call(exp.Table(db=bla_id))])
+
+    renderer(
+        "{%- set relation = api.Relation.create(schema='bla') -%}"
+        "{{ adapter.create_schema(relation) }}"
+    )
+    adapter_mock.create_schema.assert_has_calls([call(bla_id)])
+
+    renderer(
+        "{%- set relation = api.Relation.create(schema='bla') -%}"
+        "{{ adapter.drop_schema(relation) }}"
+    )
+    adapter_mock.drop_schema.assert_has_calls([call(bla_id)])
+
+    renderer(
+        "{%- set relation = api.Relation.create(schema='bla', identifier='bob') -%}"
+        "{{ adapter.drop_relation(relation) }}"
+    )
+    adapter_mock.drop_table.assert_has_calls([call(exp.Table(this=bob_id, db=bla_id))])
+
+    select_star_query: exp.Select = exp.maybe_parse("SELECT * FROM t", dialect="snowflake")
+
+    # The following call to run_query won't return dataframes and so we're expected to
+    # raise in adapter.execute right before returning from the method
+    with pytest.raises(AssertionError):
+        renderer("{{ run_query('SELECT * FROM t') }}")
+    adapter_mock.fetchdf.assert_has_calls([call(select_star_query, quote_identifiers=False)])
+
+    renderer("{% call statement('something') %} {{ 'SELECT * FROM t' }} {% endcall %}")
+    adapter_mock.execute.assert_has_calls([call(select_star_query, quote_identifiers=False)])
```

Comment:
Let's move run_query and statement to their own tests each

Reply:
Will do

Reply:
what about just moving all these normalization tests in a new one instead?

Reply:
moved them to `test_normalization` so the intent is clear